File: chat-example/src/main/java/com/hoten/servingchat/client/protocols/PeerDisconnect.java
package com.hoten.servingchat.client.protocols;

import com.hoten.servingchat.client.ConnectionToChatServerHandler;
import com.google.gson.JsonObject;
import com.hoten.servingjava.message.JsonMessageHandler;

public class PeerDisconnect extends JsonMessageHandler<ConnectionToChatServerHandler> {

    @Override
    protected void handle(ConnectionToChatServerHandler connection, JsonObject data) {
        String username = data.get("username").getAsString();
        connection.getChat().announceDisconnect(username);
    }
}

File: chat-example/src/main/csharp/ChatClient/ChatClient/Protocol/PeerDisconnect.cs
using Newtonsoft.Json.Linq;
using Serving;
using System;

namespace ChatClient.Protocol
{
    class PeerDisconnect : JsonMessageHandler<ConnectionToChatServerHandler>
    {
        protected override void Handle(ConnectionToChatServerHandler connection, JObject data)
        {
            var username = data["username"].Value<String>();
            connection.Chat.AnnounceDisconnect(username);
        }
    }
}
